### PR TITLE
Fixes roundend lag

### LINKED ISSUE #23218: Shuttle docking sound is a few seconds earlier than the actual dock.
Probably caused by the addtimer refactor and threw the timing a bit off.

### PR DESCRIPTION
**Pull request by Cyberboss**
Just sprinkle a few of these in here

:cl: Cyberboss
fix: Shuttle docking/round end shouldn't lag as much
/:cl:

Fixes #23218

### CODE REVIEW

File: code/controllers/subsystem/ticker.dm [hunk 1]
```diff
@@ -506,9 +525,16 @@ var/datum/subsystem/ticker/ticker
 			dellog += "Failures : [SSgarbage.didntgc[path]] \n"
 		world.log << dellog
 
+	CHECK_TICK
+
 	//Collects persistence features
 	SSpersistence.CollectData()
-	return 1
```
Comment (ChangelingRain) [inline code/controllers/subsystem/ticker.dm]: is this not important?

**Comment by lzimann** [comment]
i hate you

File: code/controllers/subsystem/ticker.dm (hunk 1 above)
Comment (Cyberboss) [inline code/controllers/subsystem/ticker.dm]: It's only called in one place and never checked, so no

**Comment by Cyberboss** [comment]
@lzimann You better fix this or I'll come after you

**Comment by MrStonedOne** [comment]
Add some here: https://github.com/tgstation/tgstation/blob/master/code/modules/research/message_server.dm#L294

that first while won't need any, the query can only return 1 row such that that while could (and should) be an if, but thats besides the point.

**Comment by Cyberboss** [comment]
That's called from Reboot though, should the call also be moved to declare_completion?

**Comment by MrStonedOne** [comment]
... so it is.

That didn't used to be the case. At one point it was called from declare completion.....

**Comment by optimumtact** [comment]
This doesn't fix lag, it just makes it less noticeable. Which shouldn't matter much anyway since the round is over.

Either way meh

**Comment by optimumtact** [comment]
Has this had a test merge?

**Comment by Cyberboss** [comment]
Not yet

**Comment by optimumtact** [comment]
and now?

**Comment by Cyberboss** [comment]
@MrStonedOne @lzimann, whenever you guys get a chance

**Comment by MrStonedOne** [comment]
Test merged on both servers

**Pull request merged**